src/link.zig

```diff
@@ -717,19 +717,37 @@ pub const File = struct {
         // directly, and remove this function from link.zig.
         _ = base;
         while (true) {
-            std.fs.rename(
-                cache_directory.handle,
-                tmp_dir_sub_path,
-                cache_directory.handle,
-                o_sub_path,
-            ) catch |err| switch (err) {
-                error.PathAlreadyExists => {
+            if (builtin.os.tag == .windows) {
+                // workaround windows `renameW` can't fail with `PathAlreadyExists`
```

Comment:
Might be worth adding a link to https://github.com/ziglang/zig/issues/8362 and/or https://github.com/ziglang/zig/issues/6364

```suggestion
                // workaround windows `renameW` can't fail with `PathAlreadyExists`
                // see https://github.com/ziglang/zig/issues/8362
                // TODO: https://github.com/ziglang/zig/issues/6364
```